wire-ios-data-model/Source/Model/Message/ZMMessage.m

```diff
@@ -266,7 +266,11 @@ - (void)expireWithExpirationReason:(ZMExpirationReason)expirationReason;
         self.expirationReasonCode = [NSNumber numberWithInteger:expirationReason];
     }
     [self removeExpirationDate];
-    self.conversation.hasUnreadUnsentMessage = YES;
+
+    if (self.visibleInConversation != nil) {
+        // Only warn for unsent messages if it's actually visible.
+        self.conversation.hasUnreadUnsentMessage = YES;
+    }
```

Comment:
This is a bonus fix where invisible messages (such as read receipts) that failed to send would show the "unsent message" indicator to show in the conversation on the conversation list.